src/ip.rs

```diff
@@ -0,0 +1,53 @@
+//! Handles parsing of Internet Protocol fields (shared between ipv4 and ipv6)
+
+#[derive(Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "derive", derive(Serialize, Deserialize))]
+pub enum IPProtocol {
```

Comment:
The only problem (now that I think about it) I see there is that addind a new protocol is a breaking change.  
Plus `Other` is not very well-named, because `Other(1)` would be `ICMP` ...

Reply:
How about:
- a pure `IPProtocol` enum with variant forcing (`ICMP = 1`) with something like `__Nonexhaustive` like described in rust-lang-nursery/api-guidelines#173
- using https://crates.io/crates/strum's `EnumDiscriminants` for conversion
- having a simple wrapper enum like:
```rust
pub enum IPProtonum {
    ByProto(IPProtocol),
    ByNumber(u8),
}
```
?  

Note that's it's only a quick thinking, I may need to think about it a bit more.

Reply:
I'd like to keep access to unrecognized values around since I'm currently using `EtherType::Other(0xfc00)` to match on a non-standard protocol. The only way to introduce new variants in a non-breaking way (while also allowing access to unrecognized values) would be a raw u8/u16 and constants.

Breaking changes aren't really a problem for me, so I'm fine with both solutions.